eng/Subsets.props

```diff
@@ -357,13 +357,14 @@
                              $(CoreClrProjectRoot)tools\r2rtest\R2RTest.csproj;
                              $(CoreClrProjectRoot)tools\PdbChecker\PdbChecker.csproj;
                              $(CoreClrProjectRoot)tools\AssemblyChecker\AssemblyChecker.csproj" Category="clr" Condition="'$(DotNetBuildSourceOnly)' != 'true'"/>
-    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2_publish.csproj" Category="clr" />
     <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler.Build.Tasks\ILCompiler.Build.Tasks.csproj" Category="clr" Condition="'$(NativeAotSupported)' == 'true'" />
     <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\ILCompiler\ILCompiler.csproj" Category="clr" Condition="'$(NativeAotSupported)' == 'true'" />
     <ProjectToBuild Include="$(CoreClrProjectRoot)nativeaot\BuildIntegration\BuildIntegration.proj" Category="clr" Condition="'$(NativeAotSupported)' == 'true'" />
-
     <ProjectToBuild Condition="'$(NativeAotSupported)' == 'true' and ('$(CrossBuild)' == 'true' or '$(BuildArchitecture)' != '$(TargetArchitecture)' or '$(HostOS)' != '$(TargetOS)' or '$(EnableNativeSanitizers)' != '')" Include="$(CoreClrProjectRoot)tools\aot\ILCompiler\ILCompiler_crossarch.csproj" Category="clr" />
+
     <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2_inbuild.csproj" Category="clr" />
+    <!-- skip the architectures that don't support self-contained at all -->
+    <ProjectToBuild Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2_publish.csproj" Condition="'$(TargetArchitecture)' != 'armel' and '$(TargetArchitecture)' != 'riscv64'" Category="clr" />
```

Comment:
The comment is a bit misleading. Here the problem is not about target not supporting the self-contained, but rather we don't have  community-supported platforms' SDK and LKG `Microsoft.NETCore.App.Runtime.linux-{platform}` in our CI, i.e. `riscv64`, `loongarch64`, `freebsd-{x64,arm64}`, `illumos-x64` etc.

Also, armel should use arm (via `ToolsRID`) per https://github.com/dotnet/runtime/blob/0ead293ea45c9fda243111242c17b64368ef46ff/src/coreclr/tools/Common/CommandLineHelpers.cs#L109

Reply:
Since this is going to break other community platforms. Maybe we should either exclude all community platforms (`Condition=<all official platforms>`) or keep them using the live build.

Alternatively, we can simplify the relevant parts of the infrastructure and allow community platforms to provide their hook into the install script (dotnet/install-scripts#501). This way, we won't need to keep track of the distinction between community and official versions throughout the process. For example, switching to the LKG version will be straightforward, with no exceptions; since any "bring-your-own-platform-SDK" issues would have been resolved earlier in the infra bring-up work for that platform.

Reply:
I think this should be solved by the proposed two-phase build, but I'm not quite sure of the set up right now.

Reply:
Honestly, the easiest way to fix this is to just require that community builds publish a runtime pack in their LKG toolset. If their local SDK contains a `freebsd-x64` runtime pack, it will be used during the build.

Reply:
> Honestly, the easiest way to fix this is to just require that community builds publish a runtime pack in their LKG toolset

If the CI builds for community targets were to take dependency on this, new LKG packages for community targets would have to be published in lock step with the Microsoft published ones. I do not think we want to be dealing with this moving part during LKG updates.

Reply:
I think there are options like this:

* we don't validate their build in our CI, we can let them create with same OfficialBulidId used in LKG. They are already using that: https://github.com/Thefrank/dotnet-freebsd-crossbuild/blob/13d2f5a393d64e377be96ff841a8045d66d1f80c/build.sh#L63
* we keep validating their build, and in LKG update PR, we use a command like `@dotnet-community-bot build LKG <version>` and it will trigger the same job which publishes for a given OfficialBuildId.